Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m

```diff
@@ -71,61 +71,255 @@ - (NSString *)standardSpotlightIdentifierFromActivity:(NSUserActivity *)userActi
+                  description:(NSString *)description
+            publiclyIndexable:(BOOL)publiclyIndexable
+                         type:(NSString *)type
+                 thumbnailUrl:(NSURL *)thumbnailUrl
+                     keywords:(NSSet *)keywords
+                     userInfo:(NSDictionary *)userInfo
+                     callback:(callbackWithUrl)callback {
+    [self indexContentWithTitle:title
+                    description:description
+              publiclyIndexable:publiclyIndexable
+                           type:type
+                   thumbnailUrl:thumbnailUrl
+                       keywords:keywords
+                       userInfo:userInfo
+                 expirationDate:nil
+                       callback:callback
+              spotlightCallback:nil];
+}
+
+- (void)indexContentWithTitle:(NSString *)title
+                  description:(NSString *)description
+            publiclyIndexable:(BOOL)publiclyIndexable
+                         type:(NSString *)type
+                 thumbnailUrl:(NSURL *)thumbnailUrl
+                     keywords:(NSSet *)keywords
+                     userInfo:(NSDictionary *)userInfo
+            spotlightCallback:(callbackWithUrlAndSpotlightIdentifier)spotlightCallback {
+    [self indexContentWithTitle:title
+                    description:description
+              publiclyIndexable:publiclyIndexable
+                           type:type
+                   thumbnailUrl:thumbnailUrl
+                       keywords:keywords
+                       userInfo:userInfo
+                 expirationDate:nil
+                       callback:nil
+              spotlightCallback:(callbackWithUrlAndSpotlightIdentifier)spotlightCallback];
+}
+
+//This is the final one, which figures out which callback to use, if any
+// The simpler callbackWithURL overrides spotlightCallback, so don't send both
+- (void)indexContentWithTitle:(NSString *)title
+                  description:(NSString *)description
+            publiclyIndexable:(BOOL)publiclyIndexable
+                         type:(NSString *)type
+                 thumbnailUrl:(NSURL *)thumbnailUrl
+                     keywords:(NSSet *)keywords
+                     userInfo:(NSDictionary *)userInfo
+               expirationDate:(NSDate *)expirationDate
+                     callback:(callbackWithUrl)callback
+            spotlightCallback:(callbackWithUrlAndSpotlightIdentifier)spotlightCallback {
 
     if ([BNCSystemObserver getOSVersion].integerValue < 9) {
         if (callback) {
             callback(nil, [NSError errorWithDomain:BNCErrorDomain code:BNCVersionError userInfo:@{ NSLocalizedDescriptionKey: @"Cannot use CoreSpotlight indexing service prior to iOS 9" }]);
+        } else if (spotlightCallback) {
+            spotlightCallback(nil, nil,[NSError errorWithDomain:BNCErrorDomain code:BNCVersionError userInfo:@{ NSLocalizedDescriptionKey: @"Cannot use CoreSpotlight indexing service prior to iOS 9" }]);
```

Comment:
nitpicking, and happy to defer, but could we only declare the `userInfo` dict once?